crates/uv/tests/python_pin.rs

```diff
@@ -229,6 +230,50 @@ fn python_pin_no_python() {
     "###);
 }
 
+#[test]
+fn python_pin_compatible_with_requires_python() -> anyhow::Result<()> {
+    let context: TestContext = TestContext::new_with_versions(&["3.11", "3.12"]);
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = ["iniconfig"]
+        "#,
+    )?;
+
+    uv_snapshot!(context.filters(), context.python_pin().arg("3.11"), @r###"
+        success: false
+        exit_code: 2
+        ----- stdout -----
+
+        ----- stderr -----
+        error: Pinned python version is incompatible with Requires-Python: >=3.12
+        "###);
+
+    uv_snapshot!(context.filters(), context.python_pin().arg("3.12"), @r###"
+        success: true
+        exit_code: 0
+        ----- stdout -----
+        Pinned to `3.12`
+
+        ----- stderr -----
+        "###);
```

Comment:
what if after successfully pinning, `requires-python` increased? should we show error on `uv python pin`?

Reply:
A subsequent `uv python pin` (without arguments) could display a warning that the pin isn't compatible, that seems nice.

Reply:
or showing warning that pinned python version is not installed

Reply:
and also, showing warning that pinned python version is not installed

Reply:
@T-256 if you're interested in taking that last part in a separate pull request that'd be cool

Reply:
Done :D